### PR TITLE
Allow related title changes

### PR DESCRIPTION
**Pull request by R33D3M33R**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This modification enables the change of the related products title via a filter. This was not possible before and related.php had to be edited.

### Changelog entry

Template: single-product/related.php - Add new filter `woocommerce_product_related_products_heading` for changing the Related Products heading without the need to override the template.

### CODE REVIEW

File: templates/single-product/related.php [hunk 1]
```diff
@@ -22,9 +22,12 @@
 if ( $related_products ) : ?>
 
 	<section class="related products">
-
-		<h2><?php esc_html_e( 'Related products', 'woocommerce' ); ?></h2>
-
+        <?php
+        $heading = apply_filters( 'woocommerce_product_related_heading', __( 'Related products', 'woocommerce' ) );
```
Comment (kloon) [inline templates/single-product/related.php]: Could you please change the filter name to `woocommerce_product_related_products_heading` instead?

**Review by kloon — CHANGES_REQUESTED** [review]
Hi, thanks for your contribution. Could you just change the filter name to the one suggested? Thanks!

Please also change the template version to 3.9.0 at the top of the file.

File: templates/single-product/related.php (hunk 1 above)
Comment (R33D3M33R) [reply templates/single-product/related.php]: Done.

Best Regards,
Andrej

**Review by kloon — CHANGES_REQUESTED** [review]
Almost there, can you please address the Coding Standards errors:

```
13 | ERROR | [x] Spaces must be used for mid-line alignment; tabs are not
    |       |     allowed
    |       |     (WordPress.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
 14 | ERROR | [x] Spaces must be used for mid-line alignment; tabs are not
    |       |     allowed
    |       |     (WordPress.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
 25 | ERROR | [x] Tabs must be used to indent lines; spaces are not allowed
    |       |     (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
 26 | ERROR | [x] Tabs must be used to indent lines; spaces are not allowed
    |       |     (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
 27 | ERROR | [x] Tabs must be used to indent lines; spaces are not allowed
    |       |     (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
 27 | ERROR | [x] Closing PHP tag must be on a line by itself
    |       |     (Squiz.PHP.EmbeddedPhp.ContentBeforeEnd)
 28 | ERROR | [x] Tabs must be used to indent lines; spaces are not allowed
    |       |     (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
 29 | ERROR | [x] Tabs must be used to indent lines; spaces are not allowed
    |       |     (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
 36 | ERROR | [x] Tabs must be used to indent lines; spaces are not allowed
    |       |     (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
 38 | ERROR | [ ] Overriding WordPress globals is prohibited. Found assignment
    |       |     to $GLOBALS['post']
    |       |     (WordPress.WP.GlobalVariablesOverride.Prohibited)
 38 | ERROR | [ ] Assignments must be the first block of code on a line
    |       |     (Squiz.PHP.DisallowMultipleAssignments.Found)
 40 | ERROR | [x] Closing PHP tag must be on a line by itself
    |       |     (Squiz.PHP.EmbeddedPhp.ContentBeforeEnd)
 48 | ERROR | [x] Opening PHP tag must be on a line by itself
    |       |     (Squiz.PHP.EmbeddedPhp.ContentAfterOpen)
```

**Comment by R33D3M33R** [comment]
I did all the changes for the file to address all the Coding Standards errors. I set the globals check to ignore, because it is a similar technique to the one used in other template files (for example cross-sells.php). 

It also seems that two additional files could benefit from a similar filter: cross-sells.php and up-sells.php. If this would be a useful feature, I would also like to include it in this PR.

Best regards,
Andrej

**Comment by kloon** [comment]
Thank you Andrej, generally, we only make these coding standard changes when we have something to fix or change in the file, so let's leave those files for now until there is a fix or feature that needs to be added to those files.

**Review by kloon — APPROVED** [review]
Thanks for all the changes, this is now good to merge.

**Pull request merged**